### ISSUE TITLE
Support queries by postal address

### ISSUE THREAD

**Issue by thyeggman** [ProgramEquity/amplify#399]
Currently, we only support queries by ZIP code.  Zip codes do not necessarily correlate with legislative districts.  By enabling searches by address, the Cicero API will return only the results for the legislative districts for that point.  This will give more granularity to search results for local representatives, and there won't be representatives presented outside the sender's legislative area(s).  Since we have caching already, this should mitigate the effects of increased cost due to additional granularity in results.

- [ ] Add another API to the app which accepts an address in its body
- [ ] Feed the address information to the [cicero address API](https://cicero.azavea.com/docs/#address)
- [ ] Update the Amplify App UI to call the new endopint if the search term looks like an address

These should be treated as separate tasks.  However, once the endpoint name is determined, work can start on the front end side to update the UI to call the new API.

**Comment by paramsiddharth** [comment]
> > * Query/Cache the Cicero API for the officials based on the district ID
> 
> @paramsiddharth Though I am using the location queries specified in the description of this issue, I am not using them with the legislative-district or nonlegislative-district endpoints. Which, according to my tests and involvement with the API, are the ones that return the district IDs, at least solely (the response specifically returns objects corresponding to the districts). The reason for not using these IDs is that you cannot search officials (referring to the `officials` endpoint) with the district_id alone, you need to include the location queries specified in [cicero.azavea.com/docs/#location-queries](https://cicero.azavea.com/docs/#location-queries). Without them, filtering by IDs will not work. The official resource works with location queries, including the street address. I have decided to directly implement it with it. Fetching the district ID by type (you have to make two calls for each type, in the best case scenario), along with the location query values on the bottom of the response (that the official resource also has), to then make another call to get the officials does not seem efficient for this case.
> 
> I don't think you need to update the steps in this reference, as it is clear to me now, but I am mentioning it because my implementation for this changed in accordance with my new understanding of the solution.

@ahn-nath Makes sense, Nath. I agree. The caching proposal I mentioned above is just to collect the ideas we discussed in our last meeting, so don't worry about firmly implementing it right away. It is something we plan to deal with once we are done with supporting address queries first, and I believe we definitely need a lot more planning than what we have done so far before even starting to implement the caching for the address queries. So you can leave it for now, and if you have already implemented some of the logic, you can comment it out and can resume working on it later. Meanwhile, we are just focusing on having the postal address query support with your ongoing draft PR.

@thyeggman What do you think about what Nath said above, Jacob?

**Comment by ahn-nath** [comment]
> @ahn-nath Makes sense, Nath. I agree. The caching proposal I mentioned above is just to collect the ideas we discussed in our last meeting, so don't worry about firmly implementing it right away. It is something we plan to deal with once we are done with supporting address queries first, and I believe we definitely need a lot more planning than what we have done so far before even starting to implement the caching for the address queries. So you can leave it for now, and if you have already implemented some of the logic, you can comment it out and can resume working on it later. Meanwhile, we are just focusing on having the postal address query support with your ongoing draft PR.


Hello, Param. Thanks for your response and for your support. The caching is not a problem at the moment. I already have the implementation and will ask for feedback soon. I only wanted to mention that I do not use the `legislative-districts` endpoint to query the officials by address. Likewise, I directly use the officials' endpoint with a new location query (search_street).

I will be updating my PR today and send it to review.

**Comment by paramsiddharth** [comment]
> [Manisha: Make the full address query **mandatory** for the local and school filters. _[Can be discussed more]_](https://github.com/ProgramEquity/amplify/pull/402#issuecomment-1322390800)

**Comment by paramsiddharth** [comment]
> > [Manisha: Make the full address query **mandatory** for the local and school filters. _\[Can be discussed more\]_](https://github.com/ProgramEquity/amplify/pull/402#issuecomment-1322390800)

@ahn-nath Nath has some feedback on this.
